### PR TITLE
Add the rest of the precompiles

### PR DESCRIPTION
**Pull request by norhh**
Couldn't find tests for elliptic curve operations, So did lots of mocking.
TODO: gas calculations

### CODE REVIEW

File: mythril/laser/ethereum/util.py [hunk 1]
```diff
@@ -150,3 +150,27 @@ def bytearray_to_int(arr):
     for a in arr:
         o = (o << 8) + a
     return o
+
+
+def extract_copy(
+    data: bytearray, mem: bytearray, memstart: int, datastart: int, size: int
+):
+    for i in range(size):
```
Comment (norhh) [inline mythril/laser/ethereum/util.py]: This should be changed

File: tests/laser/Precompiles/test_elliptic_mul.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+@patch("mythril.laser.ethereum.natives.bn128.multiply", return_value=1)
+@patch("mythril.laser.ethereum.natives.bn128.normalize")
+def test_ec_mul(f1, f2, f3):
+    FQ.fielf_modulus = 128
+    a = FQ(val=1)
+    f1.return_value = (a, a)
+    assert ec_mul(VECTOR_A) == [
+        0,
+        0,
+        0,
```
Comment (norhh) [inline tests/laser/Precompiles/test_elliptic_mul.py]: I blame this on black

**Comment by b-mueller** [comment]
I took the liberty of resolving the merge conflicts.

**Comment by norhh** [comment]
Awesome, Thanks @b-mueller

**Comment by b-mueller** [comment]
This PR is ancient. Is anybody up for resolving the conflicts an getting it merged?

File: mythril/laser/ethereum/natives.py [hunk 3]
```diff
@@ -120,18 +99,125 @@ def identity(data: List[int]) -> List[int]:
...
+    extract_copy(bytes_data, exp, 0, 96 + baselen, explen)
+    mod = bytearray(modlen)
+    extract_copy(bytes_data, mod, 0, 96 + baselen + explen, modlen)
+    if big_endian_to_int(mod) == 0:
+        return [0] * modlen
+    o = pow(big_endian_to_int(base), big_endian_to_int(exp), big_endian_to_int(mod))
+    return [safe_ord(x) for x in zpad(int_to_big_endian(o), modlen)]
+
+
+def ec_add(data: List[int]) -> List[int]:
```
Comment (nbanmp) [inline mythril/laser/ethereum/natives.py]: Doc strings explaining the parameters in `data` here (and the other natives) would be great.

**Pull request merged**